iceoryx_posh/source/runtime/message_queue_interface.cpp

```diff
@@ -145,24 +114,15 @@ bool MqBase::send(const MqMessage& msg) const noexcept
         return false;
     }
 
-    const size_t messageSize = static_cast<size_t>(msg.getMessage().size()) + NULL_TERMINATOR_SIZE;
-    if (messageSize > static_cast<size_t>(m_attr.mq_msgsize))
-    {
-        LogError() << "msg size of " << messageSize << "bigger than " << m_attr.mq_msgsize
-                   << ", maybe change MQ_MSG_SIZE";
-        return false;
-    }
-
-    unsigned int prio = 1;
-    if (-1 == mq_send(m_roudiMq, msg.getMessage().c_str(), messageSize, prio))
-    {
-        LogError() << "Calling mq_send() failed : " << strerror(errno);
-        return false;
-    }
-    else
-    {
-        return true;
-    }
+    auto logLengthError = [&msg](cxx::expected<posix::IpcChannelError>& error) {
+        if (error.get_error() == posix::IpcChannelError::MESSAGE_TOO_LONG)
+        {
+            const size_t messageSize =
+                static_cast<size_t>(msg.getMessage().size()) + posix::MessageQueue::NULL_TERMINATOR_SIZE;
```

Comment:
We need the IPC channel abstraction here isf we want to resuse this also for unix domain socket. This will become relevant when introducing unix domain socket for MAC (and also Windows?)